File: thousands-tv/wildcard/packages/interfaces/src/db/interfaces/userSponsoredEvent.ts
import { Types } from "mongoose";

export const USER_SPONSORED_EVENTS_TABLE_NAME = "user-sponsored-events";

export interface IUserSponsoredEvent {
    userId: Types.ObjectId;
    sponsoredEventId: Types.ObjectId;
    sponsorshipSlotId: string;
    rank?: number | null;
    wcEarned: number;
    thousandsXpEarned: number;
    support: number;
    tier: number;
    house: number;
    paidOn?: Date | null;
    claimedOn?: Date | null;
    usdcPrice: number;
    _id?: Types.ObjectId;
    __v?: number;
    createdAt?: Date;
    updatedAt?: Date;
}

File: thousands-services/IvsIdleGameShared/Models/Sponsorship/UserSponsoredEvent.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace IvsIdleGameShared.Models.Sponsorship;

[BsonIgnoreExtraElements]
public class UserSponsoredEvent
{
    [BsonId]
    [BsonElement("_id")]
    public ObjectId Id { get; set; }

    [BsonElement("userId")]
    public ObjectId UserId { get; set; }

    [BsonElement("sponsoredEventId")]
    public ObjectId SponsoredEventId { get; set; }

    [BsonElement("sponsorshipSlotId")]
    public ObjectId SponsorshipSlotId { get; set; }

    [BsonElement("usdcPrice")]
    public int UsdcPrice { get; set; } = 0;

    [BsonElement("tier")]
    public int Tier { get; set; } = 0;

    [BsonElement("house")]
    public int House { get; set; } = 0;

    [BsonElement("wcEarned")]
    public int WcEarned { get; set; } = 0;

    [BsonElement("thousandsXpEarned")]
    public int ThousandsXpEarned { get; set; } = 0;

    [BsonElement("support")]
    public int Support { get; set; } = 0;

    [BsonElement("claimedOn")]
    public DateTime? ClaimedOn { get; set; } = null;

    [BsonElement("paidOn")]
    public DateTime? PaidOn { get; set; } = null;

    [BsonElement("createdAt")]
    public DateTime CreatedAt { get; set; } = DateTime.MinValue;

    [BsonElement("updatedAt")]
    public DateTime UpdatedAt { get; set; } = DateTime.MinValue;

    [BsonElement("__v")]
    public int Version { get; set; } = 0;
}
